src/git.zsh

```diff
@@ -327,6 +482,24 @@ _fzf_complete_git_parse_completing_option() {
     esac
 }
 
+_fzf_complete_git_parse_argument() {
+    local arguments=(${(z)1})
+    local index=$2
+    local options_argument_required=(${(z)3})
+
+    local argument_position=${arguments[(ib:3:)[^-]*]}
+    local command_arguments=()
+    while (( argument_position <= ${#arguments} )); do
+        local argument=${${arguments[$(( argument_position - 1 ))]}##-##}
+        if [[ -z ${${options_argument_required##-##}[(r)$argument]} ]] || [[ ${${options_argument_required##-##}[(r)$argument]} != $argument ]]; then
+            command_arguments+=${arguments[$argument_position]}
+        fi
+        argument_position=${arguments[(ib:(( argument_position + 1 )):)[^-]*]}
+    done
+
+    echo ${command_arguments[$index]}
```

Comment:
To make use of this function more generally, this function should have ability to identify whether the argument is actually found or just an empty string. How about returing the value indicating whether the variable `command_arguments` has the specified `$index`?